src/main/java/edu/harvard/iq/dataverse/api/Admin.java

```diff
@@ -1045,8 +1045,8 @@ public Response registerDataFileAll() {
         logger.info("Starting to register  " +  count + " files. " + new Date());
         for (DataFile df : fileService.findAll()) {
             try {
-                if (df.getIdentifier() == null || df.getIdentifier().isEmpty()) {
-                    User u =findUserOrDie();
+                if ((df.getIdentifier() == null || df.getIdentifier().isEmpty())&& df.isReleased()) {
+                    User u =findAuthenticatedUserOrDie();
```

Comment:
I just spoke with @sekmiller about adding the check for `isReleased` and we agree that it's more consistent how new datasets are created and published. I'm sort of wondering why there was a change to `findUserOrDie`. Also, should these changes be made to the `registerDataFile` method above as well?

Reply:
@pdurbin - Sorry - the findAuthenticatedUser is a separate issue (which I meant to remove from this pull). Since I didn't - let me explain: 

findUserOrDie appears to first check the submitted key to see if it from a privateUrl and then call findAuthenticatedUserOrDie (and if there's no key, it returns the guest user). Since a privateUrl wouldn't work for multiple datasets and a guest user would also not be allowed, it seemed that calling findAuthenticatedUserOrDie() directly would be better. I have not checked to see what happens with the original code if no key is sent or a privateURL is used - I assume permissions gets checked somewhere along the way - but it seemed simpler to just reject early if a valid superuser apikey was not available. 

I think the same change could/should be made to registerDataFile, unless using a privateUrl was intended to be OK there. 

Feel free to edit directly or let me know and I can remove/add it to registerDataFile...

Reply:
@qqmyers thanks, I haven't been very close to this part of the code so I'm fine with others weighing in on whether the pull request needs more work or not.